File: MAHAndroidUpdater/mah-android-updater/src/main/java/com/mobapphome/mahandroidupdater/tools/LocaleUpdater.java
package com.mobapphome.mahandroidupdater.tools;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;

import java.util.Locale;

public class LocaleUpdater {
    private static void updateLocale(Context context, String language) {
        Locale locale = new Locale(language);
        Locale.setDefault(locale);

        Resources resources = context.getResources();

        Configuration configuration = resources.getConfiguration();
        configuration.locale = locale;

        resources.updateConfiguration(configuration, resources.getDisplayMetrics());
    }
}

File: library/src/main/java/com/mobapphome/mahandroidupdater/tools/LocaleUpdater.kt
package com.mobapphome.mahandroidupdater.tools

import android.content.Context
import android.content.res.Configuration
import android.content.res.Resources

import java.util.Locale

object LocaleUpdater {
    private fun updateLocale(context: Context, language: String) {
        val locale = Locale(language)
        Locale.setDefault(locale)

        val resources = context.resources

        val configuration = resources.configuration
        configuration.locale = locale

        resources.updateConfiguration(configuration, resources.displayMetrics)
    }
}